src/libServer/Server.cpp

```diff
@@ -193,69 +212,85 @@ Json::Value Server::CreateTransaction(const Json::Value& _json) {
 
     if (num_shards > 0) {
       unsigned int shard = Transaction::GetShardIndex(fromAddr, num_shards);
-      if (tx.GetData().empty() || tx.GetToAddr() == NullAddress) {
-        if (tx.GetData().empty() && tx.GetCode().empty() &&
-            tx.GetToAddr() != NullAddress) {
+      switch (GetTransactionType(tx)) {
+        case NON_CONTRACT:
           if (!ARCHIVAL_LOOKUP) {
             m_mediator.m_lookup->AddToTxnShardMap(tx, shard);
           } else {
-            m_mediator.m_lookup->AddToTxnShardMap(tx, 0);
+            m_mediator.m_lookup->AddToTxnShardMap(
+                tx, SEND_TYPE::ARCHIVAL_SEND_SHARD);
           }
           ret["Info"] = "Non-contract txn, sent to shard";
           ret["TranID"] = tx.GetTranID().hex();
-        } else if (!tx.GetCode().empty() && tx.GetToAddr() == NullAddress) {
+          return ret;
+          break;
```

Comment:
The break-after-return pattern repeats several times in this switch statement. Generally speaking, it's not clean as `break` won't be executed anyway. If `break` is only used to remind people, then perhaps we should not mix the use of break and return.